src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SpacingFormattingRule.cs

```diff
@@ -330,23 +330,88 @@ public override AbstractFormattingRule WithOptions(AnalyzerConfigOptions options
                 }
 
                 // Force a space between * and the calling convention
-                if (currentKind == SyntaxKind.IdentifierToken && previousKind == SyntaxKind.AsteriskToken)
+                if (previousKind == SyntaxKind.AsteriskToken && currentParentKind == SyntaxKindEx.FunctionPointerCallingConvention)
                 {
-                    return CreateAdjustSpacesOperation(1, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
+                    switch (currentKind)
+                    {
+                        case SyntaxKind.IdentifierToken:
+                        case SyntaxKindEx.ManagedKeyword:
+                        case SyntaxKindEx.UnmanagedKeyword:
+                            return CreateAdjustSpacesOperation(1, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
+                    }
+                }
```

Comment:
If we just checked that the currentToken is the FunctionPointerCallingConvention's ManagedSpecifier then we can remove this block, right? And we're having to do this because of the funkiness around not being able to reference new syntax node types in our code?

Reply:
Yeah, we could accomplish it that way.

---
In reply to: [451066787](https://github.com/dotnet/roslyn/pull/45684#discussion_r451066787) [](ancestors = 451066787)

Reply:
@333fred If you could either do that work or just file a bug to do it, I think I'd feel better since this then is a bit more flexible around precisely what we might get out of the parser.

Reply:
Filed https://github.com/dotnet/roslyn/issues/45897.